packages/devextreme/testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -504,6 +504,26 @@ QUnit.module('interaction via swipe', {
         });
     });
 
+    QUnit.test('when all items are hidden selectedIndex should be initialized to 0 after swipe', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: false },
+                { text: '2', visible: false },
+                { text: '3', visible: false },
+            ],
+            selectedIndex: 2,
+            loop: true,
+            swipeEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $itemContainer = $multiView.find(`.${MULTIVIEW_ITEM_CONTAINER_CLASS}`);
+        const pointer = pointerMock($multiView);
+
+        pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1);
+        assert.strictEqual(position($itemContainer), 0, 'container did not move');
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is not changed');
+    });
```

Comment:
[broken behavior]
We cannot write such a test because it breaks the page if it's red. If it becomes red one day, it'll be impossible to see the test report / debug / find out the reason because test will be just unresponsive => this test is not only not helpful but also harmful.
For testing your scenario you can mock `_isVisible` method to check how many times it's called. Actually that's also not very good approach because it tests implementation and if we refactor it some day the test can become red or worse evergreen. But that's the easiest approach.

```javascript
        const isItemVisibleStub = sinon
            .stub(instance, '_isItemVisible')
            .callsFake(() => {
                if(itemVisibleStub.callCount === 20) {
                    assert.ok(false, 'infinite loop');
                    return true;
                }
                return false;
            });

        pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1);

        assert.ok(true, 'no infinite loop');
```

Reply:
[563cea3](https://github.com/DevExpress/DevExtreme/pull/28299/commits/563cea32b40366930aa50fa71d7ae64a401c7b3a)